liquibase-standard/src/main/java/liquibase/command/core/CalculateChecksumCommandStep.java

```diff
@@ -20,22 +31,41 @@ public class CalculateChecksumCommandStep extends AbstractCommandStep {
     protected static final String[] COMMAND_NAME = {"calculateChecksum"};
 
     public static final CommandArgumentDefinition<String> CHANGELOG_FILE_ARG;
-    public static final CommandArgumentDefinition<String> CHANGESET_IDENTIFIER_ARG;
+    public static final CommandArgumentDefinition<String> CHANGESET_PATH_ARG;
+
+    public static final CommandArgumentDefinition<String> CHANGESET_ID_ARG;
+
+    public static final CommandArgumentDefinition<String> CHANGESET_AUTHOR_ARG;
 
     public static final CommandResultDefinition<CheckSum> CHECKSUM_RESULT;
 
-    protected static final int CHANGESET_ID_NUM_PARTS = 3;
-    protected static final int CHANGESET_ID_AUTHOR_PART = 2;
-    protected static final int CHANGESET_ID_CHANGESET_PART = 1;
-    protected static final int CHANGESET_ID_CHANGELOG_PART = 0;
+    public static final CommandArgumentDefinition<String> CHANGESET_IDENTIFIER_ARG;
 
 
     static {
         CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
         CHANGELOG_FILE_ARG = builder.argument(CommonArgumentNames.CHANGELOG_FILE, String.class).required()
-                .description("The root changelog file").build();
-        CHANGESET_IDENTIFIER_ARG = builder.argument("changesetIdentifier", String.class).required()
-                .description("Changeset ID identifier of form filepath::id::author").build();
+                                    .description("The root changelog file").build();
+
+        CHANGESET_IDENTIFIER_ARG = builder.argument("changeSetIdentifier", String.class)
```

Comment:
Hi @JulienMa94, I have been reviewing this PR yesterday and talked about it with the team and we have a few more things we might need to change. One of these is right now we are specifying the four arguments (`CHANGESET_IDENTIFIER_ARG`, `CHANGESET_PATH_ARG`, `CHANGESET_ID_ARG` and `CHANGESET_AUTHOR_ARG`) all defined as required. And I think this could be a bit ambiguous. What one of my team members suggest is we should make the four of them optional and then add a bit of logic in the `run()` method to control which of these arguments we will use. 

Doing the mentioned update will require us to update some of our command definition tests, plus in order to avoid confusion for other users I think we can add some examples like the ones below to command help. We can do this by calling `commandDefinition.setHelpFooter()` in the `adjustCommandDefinition()` method.

```
liquibase calculateCheckSum --changeSetIdentifier myFile::myId::myAuthor
```
 and 
```
liquibase calculatCheckSum --changesetId myId --changesetAuthor myAuthor --changesetPath myPath
```
If you need any help I'm here to help/assist you as you need, or I can even help doing these changes if you don't have any time constraints. 

Thanks,
Daniel.